### DISCUSSION TITLE
Share habits and progress with friends

### DISCUSSION THREAD

**Discussion by shamrt** [iSoron/uhabits#223] [Feature Requests]
There's evidence showing that public accountability helps with goal follow through. Would you be open to adding a friend's feature so that users can share their habit's with other friends who use the app? The goal would be to track each other's progress, and perhaps compete in challenges.

**Comment by licaon-kter** [comment]
The shared data would be what: short text with progress or picture of graph or ?

**Comment by iSoron** [comment]
Thank you for the suggestion. Once users are able to synchronize their data across devices #104, we could also allow them to "share" some of their habits with their friends. What I mean by this is that their friends would be able to see (but not modify) part of the data, and would be able to track the user's progress in real time. Is this what you had in mind?

Since this requires device sync, a feature that is not available yet, I am putting it on-hold.

**Comment by deutrino** [comment]
Ideas:

* implement a simple way of sharing using only OS intents 
* the ability to share a canned phrase, like "I just achieved a new personal best in the habit I'm cultivating! 14 days in a row of [habit]" 
* the ability to choose a graph - like the strength graph, or a streak, etc - and export an image (with a little canned text, if that's allowed by the OS) to apps that can accept shared images

As for how the user can choose to use this feature: beyond adding little share icons where appropriate in the UI, offer the user to share when they reach certain milestones, such as streaks, or surpassing a certain strength level, or even more esoteric things like if a habit was often left undone on the weekend and then the user has been doing it more often on the weekends for a while.

Reasons I like this:

* It's more positive reinforcement for the user. People like praise and they like recognition. The ability to share on demand gives an opportunity to self-validate and get recognition for doing so; meanwhile, being offered to share upon achieving some milestone functions as praise, even if the user chooses not to share 🙂
* using intents pushes all the mechanics beyond the initial sharing onto whatever other apps the user is using - much less coding required for something that people might like a lot, and much less thinking about privacy implications etc

**Reply by nimeshkadecha** [reply]
we can use  "I just achieved a new personal best in the habit I'm cultivating! 14 days in a row of [habit]" as an notification to motivate the inactive or the users who are behind in streak...

**Comment by lashoun** [comment]
Another idea: enable the user to send a notification to a given friend when the user has just checked a given habit, and reciprocally, enable the user to be notified when a given friend has checked a given habit.

**Reply by nimeshkadecha** [reply]
Really Great idea....
